public/app/plugins/datasource/grafana-azure-monitor-datasource/types/query.ts

```diff
@@ -67,7 +68,7 @@ export interface AzureMetricQuery {
  */
 export interface AzureLogsQuery {
   query?: string;
-  resultFormat?: string;
+  resultFormat?: AzureResultFormat;
```

Comment:
HMMMMMMMMMMMMMMMMM I'm not sure if i actually like this change or not, though I understand and can spot the inconsistencies between this and the dropdown options. Again, I'll have a think about.

The frontend doesn't actually care what this value is, so I think i would prefer it to be typed as `string` everywhere and just drop the `AzureResultFormat` type. It's needless strictness that isn't actually enforced at the boundaries.

Reply:
your wish is my command 🪄